### PR TITLE
Add webex as destination

### PR DESCRIPTION
**Pull request by anirudhbagri**
## What type of PR is this? (check all applicable)

- [ ] New Alert Destination - Webex

## Description
Added Webex as an alert destination

### CODE REVIEW

**Comment by guidopetri** [comment]
@anirudhbagri , thanks for the PR! We've updated a lot of things now that we're Community-driven so - if you're still interested in getting this merged - would you mind rebasing off master to re-run the CI, as well as updating merge conflicts?

We're trying to clean up our PR todo list, so if you're not interested, that's fine - we'll close the PR in about a week if we don't hear back. If you're interested in reopening the PR afterwards, we would also very much welcome that.

**Comment by justinclift** [comment]
After fixing the merge conflict through the web interface, it was failing the formatting check test.

When fixing that though (`make format`), it was unable to complete fully due to other errors:

```
redash/destinations/webex.py:5:1: F403 'from redash.destinations import *' used; unable to detect undefined names
redash/destinations/webex.py:8:13: F405 'BaseDestination' may be undefined, or defined from star imports: redash.destinations
redash/destinations/webex.py:145:1: F405 'register' may be undefined, or defined from star imports: redash.destinations
```

Those would need to be fixed (by @anirudhbagri probably?), then this can be merged. :smile:

**Comment by guidopetri** [comment]
#5230 had some changes that are relevant to this PR as well, and it seems we don't have any tests on this branch yet. @anirudhbagri any interest in continuing development here? I saw you merged the master branch in about a week ago, so I don't want to step on your toes. :)

File: client/app/assets/images/destinations/webex.png
Comment (guidopetri) [inline client/app/assets/images/destinations/webex.png:1]: This image has a white background (as opposed to transparent) and is the wrong size compared to the other images (it's 512x512 and the others are 256x256).

File: redash/destinations/webex.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    "type": "string",
+                    "title": "People (comma-separated)",
+                },
+                "to_room_ids": {
+                    "type": "string",
+                    "title": "Rooms (comma-separated)",
+                },
+                "organization": {"type": "string", "title": "Organization"},
+            },
+            "secret": ["webex_bot_token"],
```
Comment (guidopetri) [inline redash/destinations/webex.py:24]: We should make the Webex bot token required.

File: redash/destinations/webex.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                                        },
+                                    ],
+                                },
+                                {
+                                    "type": "Column",
+                                    "width": 1,
+                                    "items": [
+                                        {
+                                            "type": "Image",
+                                            "url": "https://cdn.pixabay.com/photo/2016/09/16/09/20/alarm-1673577_960_720.png",
```
Comment (guidopetri) [inline redash/destinations/webex.py:94]: I don't think we should use this CDN's image directly like this (even if it seems royalty-free). They don't require credits as per the [ToS](https://pixabay.com/service/terms/) but we should probably credit them anyway if we want to use their image.

File: redash/destinations/webex.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                )
+                logging.warning(resp.text)
+                if resp.status_code != 200:
+                    logging.error(
+                        "Webex send ERROR. status_code => {status}".format(
+                            status=resp.status_code
+                        )
+                    )
+            except Exception:
+                logging.exception("Webex send ERROR.")
```
Comment (guidopetri) [inline redash/destinations/webex.py:136]: We should log what exception we get here if we run into one.

File: redash/destinations/webex.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                "organization": {"type": "string", "title": "Organization"},
+            },
+            "secret": ["webex_bot_token"],
+        }
+
+    @classmethod
+    def icon(cls):
+        return "fa-webex"
+
+    def notify(self, alert, query, user, new_state, app, host, options):
```
Comment (guidopetri) [inline redash/destinations/webex.py:31]: There's a new kwarg here for `metadata` that we have to add before this works.

**Comment by guidopetri** [comment]
Missing a test now, but otherwise looks good.

**Comment by justinclift** [comment]
Cool.  Should we merge this now, or are you still wanting to do stuff with it? :smile:

**Comment by guidopetri** [comment]
Still adding a test ;) it should pass now (it did on my local), so we can merge unless you think otherwise.

**Comment by justinclift** [comment]
Heh Heh Heh.  I'll leave it alone and let you merge it when you're happy with it. :smile:

**Comment by guidopetri** [comment]
@anirudhbagri thanks again for your contribution :)

**Pull request merged**